Modules/PHOS/src/DigitsQcTask.cxx

```diff
@@ -0,0 +1,427 @@
+    if (modNcell[mod] > 0) {
+      modTotCellEnergy[mod] /= modNcell[mod];
+    }
+    mCellMeanEnergy[mod]->Fill(modTotCellEnergy[mod]);
+    mCellN[mod]->Fill(modNcell[mod] * mAcceptanceCorrection[mod]);
+  }
+
+  // //Mark fake triggers: trigger without cluster
+  // for(Int_t ix=1; ix<=64;ix++){
+  //   for(Int_t iz=1; iz<=56; iz++){
+  //     if(mTrig[mod][ix][iz]){ //there is fired trigger without cluster
+  //           mFakeTrig[mod]->Fill(ix+0.5,iz+0.5,1.) ;
+  //     }
+  //   }
+  // }
+}
+//___________________________________________
+void DigitsQcTask::publishPhysicsObjects()
+{
+  //Prepare objects for physics runs
+
+  for (int mod = 1; mod < nMod; mod++) {
+    // if(!mTRGoccupancy[mod] || !getObjectsManager()->getObject(((moTRGoccupancy[mod]->GetName()).c_str())) {
+    //   mTRGoccupancy[mod] = new TH2F(Form("TRGoccup%d",mod), Form("Number of fired triggers per channel in mod %d",mod), 64, 0., 64., 56, 0., 56.);
+    // }
+    // else{
+    //   mTRGoccupancy[mod]->Reset();
+    // }
+    // if(!mTRGsignal[mod] || !getObjectsManager()->getObject(((mTRGsignal[mod]->GetName()).c_str())) {
+    //   mTRGsignal[mod] = new TH2F(Form("TRGsignal%d",mod), Form("Sum of amplitudes of fired triggers per channel in mod %d",mod), 64, 0., 64., 56, 0., 56.);
+    // }
+    // else{
+    //   mTRGsignal[mod]->Reset();
+    // }
+
+    // if(!mL1occupancy[mod] || !getObjectsManager()->getObject(((moL1occupancy[mod]->GetName()).c_str())) {
+    //   mL1occupancy[mod]=new TH2F(Form("L1occup%d",mod), Form("Number of fired L1 Lo threshold triggers per channel in mod %d",mod), 64, 0., 64., 56, 0., 56.);
+    // }
+    // else{
+    //   mL1occupancy[mod]->Reset();
+    // }
+    // if(!mFakeTrig[mod] || !getObjectsManager()->getObject(((moFakeTrig[mod]->GetName()).c_str())) {
+    //   mFakeTrig[mod]=new TH2F(Form("TRGfake%d",mod), Form("Number of fake triggers fired in channel in mod %d",mod), 64, 0., 64., 56, 0., 56.);
+    // }
+    // else{
+    //   mFakeTrig[mod]->Reset();
+    // }
+    if (!mTimeE[mod]) {
+      mTimeE[mod] = new TH2F(Form("TimeM%d", mod), Form("Cell time vs cell energy in module %d", mod), 100, 0., 20., 200, -300.e-9, 300.e-9);
+      mTimeE[mod]->GetXaxis()->SetTitle("E_{digit} (GeV)");
+      mTimeE[mod]->GetYaxis()->SetTitle("#tau_{digit} (s)");
+      try {
+        getObjectsManager()->startPublishing(mTimeE[mod]);
+      } catch (boost::exception& e) {
+        getObjectsManager()->stopPublishing(Form("TimeM%d", mod));
+        getObjectsManager()->startPublishing(mTimeE[mod]);
+      }
+    } else {
+      mTimeE[mod]->Reset();
+    }
+
+    if (!mCellN[mod]) {
+      mCellN[mod] = new TH1F(Form("CellMeanNM%d", mod), Form("Average number of cells in module %d", mod), 1000, 0., 1000.);
+      mCellN[mod]->GetXaxis()->SetTitle("N_{cell}/event");
+      mCellN[mod]->GetYaxis()->SetTitle("dN_{events}/dN_{cell}");
+      try {
+        getObjectsManager()->startPublishing(mCellN[mod]);
```

Comment:
I am not sure to follow the logic here. If the publication of `mCellN[mod]` fails, then we stop publishing the Mean but try again to publish `mCellN[mod]` ?

Reply:
What I see in code of Manager::publish(), it checks if this object already published and sends exception if positive. Then I stop publishing of old object and start publishing of new one. I do not see other way to check if object is published: it anyway goes through exceptions.

Reply:
Hi, Thanks for this very prompt reply. I misunderstood indeed, this is correct. I am adding a method "isBeingPublished" that should help.